packages/render-fidelity-tools/src/common.ts

```diff
@@ -174,80 +168,79 @@ export class ImageComparator {
-
-        if (generateVisuals) {
-          const deltaIntensity =
-              Math.round(255 * thresholdDelta / MAX_COLOR_DISTANCE);
-
-          maximumDeltaIntensity =
-              Math.max(deltaIntensity, maximumDeltaIntensity);
-
-          this.drawPixel(
-              blackWhiteImage!,
-              position,
-              exactlyMatched,
-              exactlyMatched,
-              exactlyMatched);
-          this.drawPixel(
-              deltaImage!,
-              position,
-              255,
-              255 - deltaIntensity,
-              255 - deltaIntensity);
-        }
+        const deltaIntensity =
+            Math.round(255 * thresholdDelta / MAX_COLOR_DISTANCE);
+
+        maximumDeltaIntensity = Math.max(deltaIntensity, maximumDeltaIntensity);
+
+        this.drawPixel(
+            blackWhiteImage!,
+            position,
+            exactlyMatched,
+            exactlyMatched,
+            exactlyMatched);
+        this.drawPixel(
+            deltaImage!,
+            position,
+            255,
+            255 - deltaIntensity,
+            255 - deltaIntensity);
       }
     }
 
-    if (generateVisuals) {
-      for (let y = 0; y < height; ++y) {
-        for (let x = 0; x < width; ++x) {
-          const index = y * width + x;
-          const position = index * COMPONENTS_PER_PIXEL;
-          const absoluteDeltaIntensity = 255 - deltaImage![position + 1];
-          const relativeDeltaIntensity = Math.round(
-              255 - 255 * (absoluteDeltaIntensity / maximumDeltaIntensity));
-
-          this.drawPixel(
-              deltaImage!,
-              position,
-              255,
-              relativeDeltaIntensity,
-              relativeDeltaIntensity);
-        }
+    for (let y = 0; y < height; ++y) {
+      for (let x = 0; x < width; ++x) {
+        const index = y * width + x;
+        const position = index * COMPONENTS_PER_PIXEL;
+        const absoluteDeltaIntensity = 255 - deltaImage![position + 1];
+        const relativeDeltaIntensity = Math.round(
+            255 - 255 * (absoluteDeltaIntensity / maximumDeltaIntensity));
+
+        this.drawPixel(
+            deltaImage!,
+            position,
+            255,
+            relativeDeltaIntensity,
+            relativeDeltaIntensity);
       }
     }
 
-    const mismatchingPixels = this.imagePixels - matched;
-
-    const mismatchingAverageDistanceRatio = mismatchingPixels > 0 ?
-        mismatchingSum / mismatchingPixels / MAX_COLOR_DISTANCE :
-        0;
-    const averageDistanceRatio = sum / this.imagePixels / MAX_COLOR_DISTANCE;
-    const rmsDistanceRatio =
-        Math.sqrt(squareSum / this.imagePixels) / MAX_COLOR_DISTANCE;
-
     return {
-      analysis: {
-        matchingRatio: matched / this.imagePixels,
-        averageDistanceRatio,
-        mismatchingAverageDistanceRatio,
-        rmsDistanceRatio
-      },
       imageBuffers: {
         delta: deltaImage ? deltaImage.buffer as ArrayBuffer : null,
```

Comment:
At first glance it looks like deltaImage can never be null but maybe I'm missing something.  I'm also surprised that its buffer field needs to be casted to `ArrayBuffer`.

Reply:
Thank you for pointing this out. Although I just copied this code from original project, i found it wired too after you mentioned. i removed the post-fix and casting to ArrayBuffer, no compiling errors. Thank you for pointing out this !